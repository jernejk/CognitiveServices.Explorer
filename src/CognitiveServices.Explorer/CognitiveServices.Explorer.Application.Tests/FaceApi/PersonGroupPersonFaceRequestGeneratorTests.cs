using CognitiveServices.Explorer.Application.FaceApi;
using FluentAssertions;
using Snapshooter.Xunit;
using System;
using Xunit;

namespace CognitiveServices.Explorer.Application.Tests.FaceApi
{
    public class PersonGroupPersonFaceRequestGeneratorTests
    {
        [Fact]
        public void ShouldGenerateAddBinary()
        {
            PersonGroupPersonFaceRequestGenerator
                .Add("default", Guid.Empty.ToString(), new byte[0])
                .Should()
                .MatchSnapshot();
        }

        [Fact]
        public void ShouldGenerateAddUrl()
        {
            PersonGroupPersonFaceRequestGenerator
                .Add("default", Guid.Empty.ToString(), "https://image.url")
                .Should()
                .MatchSnapshot();
        }

        [Fact]
        public void ShouldGenerateAddUrlWithAttributes()
        {
            PersonGroupPersonFaceRequestGenerator
                .Add("default", Guid.Empty.ToString(), "https://image.url", FaceRequestGenerator.DefaultDetectionModel, @"{""image"":""data""}")
                .Should()
                .MatchSnapshot();
        }

        [Fact]
        public void ShouldGenerateGet()
        {
            PersonGroupPersonFaceRequestGenerator
                .Get("default", Guid.Empty.ToString(), new Guid("4684f729-57d2-4512-907d-b9e0f0213bdb").ToString())
                .Should()
                .MatchSnapshot();
        }

        [Fact]
        public void ShouldGenerateUpdate()
        {
            PersonGroupPersonFaceRequestGenerator
                .Update("default", Guid.Empty.ToString(), new Guid("4684f729-57d2-4512-907d-b9e0f0213bdb").ToString(), @"{}")
                .Should()
                .MatchSnapshot();
        }

        [Fact]
        public void ShouldGenerateDelete()
        {
            PersonGroupPersonFaceRequestGenerator
                .Delete("default", Guid.Empty.ToString(), new Guid("4684f729-57d2-4512-907d-b9e0f0213bdb").ToString())
                .Should()
                .MatchSnapshot();
        }
    }
}
