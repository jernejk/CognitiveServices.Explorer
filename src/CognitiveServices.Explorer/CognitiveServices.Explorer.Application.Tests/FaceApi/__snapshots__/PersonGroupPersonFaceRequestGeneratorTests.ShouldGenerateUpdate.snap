{
  "RequestName": null,
  "Headers": {},
  "RelativePath": "face/v1.0/persongroups/default/persons/00000000-0000-0000-0000-000000000000/persistedFaces/4684f729-57d2-4512-907d-b9e0f0213bdb",
  "Queries": null,
  "HttpMethod": "PATCH",
  "Body": "{\"userData\":\"{}\"}",
  "BinaryContent": null,
  "ContentType": "application/json",
  "ExpectedHtppCode": 200,
  "TokenHeaderName": "Ocp-Apim-Subscription-Key",
  "Cost": {
    "ServiceName": "FaceApi",
    "Cost": 1.0,
    "Unit": "tr"
  },
  "CognitiveServiceDoc": "https://westus.dev.cognitive.microsoft.com/docs/services/563879b61984550e40cbbe8d/operations/563879b61984550f30395243"
}
